Bourreau/spec/boutiques/boutiques_tester_spec.rb

```diff
@@ -0,0 +1,389 @@
+      # This is currently not happening, possibly due to some versioning issues
+      before(:each) do
+        # Destroy any pre-existing userfiles
+        Userfile.all.each{ |uf| uf.destroy }
+        # Create a mock task required output file
+        @fname = DefReqOutName
+        FileUtils.touch(@fname)
+        # Helper method for getting filetype classes
+        ftype = lambda { |fname| Userfile.suggested_file_type(fname) || SingleFile }
+        @userfileClass = ftype.(@fname)
+        # Get the schema and json descriptor
+        descriptor     = File.join(__dir__, TestScriptDescriptor)
+        # Generate a new task class via the Boutiques framework and integrate it into cbrain
+        @boutiquesTask = SchemaTaskGenerator.generate(SchemaTaskGenerator.default_schema, descriptor)
+        @boutiquesTask.integrate
+        # Instantiate an object of the new class type
+        @task          = CbrainTask::BoutiquesTest.new
+        # Destroy any prior existing data providers (so we use a clean, lone one)
+        DataProvider.all.each { |dp| dp.destroy }
+        # Create a local data_provider to hold our files
+        @provider      = FlatDirLocalDataProvider.new({ :online => true, :read_only => false, :remote_dir => '.' })
+        @provider.id, @provider.name = DPID, 'test_provider'
+        @provider.user_id, @provider.group_id = UID, GID
+        @provider.save!
+        # Add a local input file to the data provider (allows smarter lookup in userfile_exists)
+        @file_c, @ft   = 'c', ftype.(@file_c)
+        newFile = @task.safe_userfile_find_or_new(@ft, name: @file_c, data_provider_id: DPID, user_id: UID, group_id: GID)
+        newFile.save!
+        # Fill in data necessary for the task to check for and save the output file '@fname'
+        @task.params   = {:ro => @fname, :interface_userfile_ids => [Userfile.all.first.id]}
+        @task.user_id, @task.group_id = UID, GID
+        # Generate a simulated exit file, as if the task had run
+        @simExitFile   = @task.exit_cluster_filename
+        FileUtils.touch( @simExitFile )
+        # The basic properties for the required output file
+        @reqOutfileProps = {:name => @fname, :data_provider_id => @provider.id}
+        # Optional output file properties
+        @optOutFileName  = OptOutName
+        @optFileClass    = ftype.( @optOutFileName )
+      end
+
+      # Clean up after each test
+      after(:each) do
+        # Delete any generated output files
+        destroyOutputFiles
+        # Also need to get rid of the (simulated) exit file
+        fename = '.qsub.exit.BoutiquesTest.-1'
+        File.delete( fename ) if File.exists?( fename )
+        # Destroy the registered userfiles and the data_provider, so as not to affect downstream tests
+        # Necessary for tests that create and/or register userfiles in the it block
+        Userfile.all.each{ |uf| uf.destroy }
+        DataProvider.all.each { |dp| dp.destroy }
+      end
```

Comment:
It seems useless to destroy the Userfile and the DataProvider, FactoryGirl already take care of that.